### PR TITLE
feat: OPX-3.1.0-dev2

### PR DESCRIPTION
**Pull request by GarrickHe**
* Feature: Buffer Statistics

Signed-off-by: Garrick He <garrick_he@dell.com>

### CODE REVIEW

**Comment by GarrickHe** [comment]
Pull-request updated, HEAD is now 0bc7501455e619439579c4ce4289ef60ed204c3f

File: src/nas_ndi_qos_buffer_pool.cpp [hunk 1]
```diff
@@ -523,11 +567,31 @@ uint_t ndi_qos_get_shadow_buffer_pool_list(npu_id_t npu_id,
  * @param list of buffer_pool counter types to clear
  * @param number of buffer_pool counter types specified
  * return standard error
- */
+ * @deprecated since 7.7.0
```
Comment (rakeshdatta) [inline src/nas_ndi_qos_buffer_pool.cpp]: since 7.7.0+opx1?

File: src/nas_ndi_qos_priority_group.cpp [hunk 2]
```diff
@@ -257,15 +279,16 @@ static void _fill_counter_stat_by_type(sai_ingress_priority_group_stat_t type, u
...
 /**
  * This function gets the priority_group statistics
  * @param ndi_port_id
  * @param ndi_priority_group_id
  * @param list of priority_group counter types to query
  * @param number of priority_group counter types specified
  * @param[out] counter stats
-  * return standard error
+ * return standard error
+ * @deprecated since 7.7.0
```
Comment (rakeshdatta) [inline src/nas_ndi_qos_priority_group.cpp]: since 4.8.0+opx1?

File: src/nas_ndi_qos_priority_group.cpp [hunk 3]
```diff
@@ -309,18 +332,106 @@ t_std_error ndi_qos_get_priority_group_stats(ndi_port_t ndi_port_id,
...
+}
+
 /**
  * This function clears the priority_group statistics
  * @param ndi_port_id
  * @param ndi_priority_group_id
  * @param list of priority_group counter types to clear
  * @param number of priority_group counter types specified
  * return standard error
+ * @deprecated since 7.7.0
```
Comment (rakeshdatta) [inline src/nas_ndi_qos_priority_group.cpp]: since 7.7.0+opx1?

File: src/nas_ndi_qos_queue.cpp [hunk 4]
```diff
@@ -618,13 +634,39 @@ t_std_error ndi_qos_get_queue_stats(ndi_port_t ndi_port_id,
  * @param list of queue counter types to query
  * @param number of queue counter types specified
  * @param[out] counters: stats will be stored in the same order of the counter_ids
-  * return standard error
+ * return standard error
+ * @deprecated since 7.7.0
```
Comment (rakeshdatta) [inline src/nas_ndi_qos_queue.cpp]: since 4.8.0+opx1?

File: src/nas_ndi_qos_queue.cpp [hunk 5]
```diff
@@ -567,7 +580,10 @@ static void _fill_counter_stat_by_type(sai_queue_stat_t type, uint64_t val,
  * @param list of queue counter types to query
  * @param number of queue counter types specified
  * @param[out] counter stats
-  * return standard error
+ * return standard error
+ * @deprecated since 7.7.0
```
Comment (rakeshdatta) [inline src/nas_ndi_qos_queue.cpp]: since 7.7.0+opx1?

File: src/nas_ndi_qos_queue.cpp [hunk 6]
```diff
@@ -676,14 +719,17 @@ t_std_error ndi_qos_get_queue_statistics(ndi_port_t ndi_port_id,
...
+
 /**
  * This function clears the queue statistics
  * @param ndi_port_id
  * @param ndi_queue_id
  * @param list of queue counter types to clear
  * @param number of queue counter types specified
  * return standard error
- */
+ * @deprecated since 7.7.0
```
Comment (rakeshdatta) [inline src/nas_ndi_qos_queue.cpp]: since 7.7.0+opx1?

File: debian/changelog [hunk 7]
```diff
@@ -1,3 +1,9 @@
+opx-nas-ndi (4.8.0+opx1) unstable; urgency=medium
+
+  * Feature: Buffer Statistics
```
Comment (rakeshdatta) [inline debian/changelog:3]: Plz add the detailed changelog as discussed offline

**Review by rakeshdatta — CHANGES_REQUESTED** [review]

**Review by atanu-mandal — DISMISSED** [review]

**Comment by GarrickHe** [comment]
Pull-request updated, HEAD is now 8c768b88c9807b245a18da72b3dc5b81f07d2dce

**Review by rakeshdatta — APPROVED** [review]

**Review by atanu-mandal — APPROVED** [review]

**Pull request merged**